### PR TITLE
3790 - Support CrateDB database plugin

### PR DESCRIPTION
**Pull request by xdelox**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:
- [x] Run `mvn clean install apache-rat:check` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
- [x] If you have a group of commits related to the same change, please squash your commits into one and force push your branch using `git rebase -i`.
- [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable.

To make clear that you license your contribution under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

- [x] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
- [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by bamaer** [comment]
Looks good, @xdelox. The only thing that seems missing is the doc page.

**Comment by xdelox** [comment]
> Looks good, @xdelox. The only thing that seems missing is the doc page.

@bamaer I think doc is ok now

File: plugins/databases/cratedb/src/test/java/org/apache/hop/databases/cratedb/CrateDBDatabaseMetaTest.java [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hop.databases.cratedb;
+
+import org.apache.hop.core.database.DatabaseMeta;
+import org.apache.hop.core.row.value.*;
```
Comment (hansva) [inline plugins/databases/cratedb/src/test/java/org/apache/hop/databases/cratedb/CrateDBDatabaseMetaTest.java:20]: StarImports will fail with our new checkstyle configuration

File: plugins/databases/cratedb/src/test/java/org/apache/hop/databases/cratedb/CrateDBDatabaseMetaTest.java [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
```
Comment (hansva) [inline plugins/databases/cratedb/src/test/java/org/apache/hop/databases/cratedb/CrateDBDatabaseMetaTest.java:24]: StarImports will fail with our new checkstyle configuration

File: plugins/databases/cratedb/src/test/java/org/apache/hop/databases/cratedb/CrateDBValueMetaBaseTest.java [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+import org.apache.hop.core.logging.IHopLoggingEventListener;
+import org.apache.hop.core.plugins.PluginRegistry;
+import org.apache.hop.core.row.IValueMeta;
+import org.apache.hop.core.row.value.ValueMetaBase;
+import org.apache.hop.core.row.value.ValueMetaFactory;
+import org.apache.hop.core.row.value.ValueMetaPluginType;
+import org.apache.hop.core.variables.IVariables;
+import org.apache.hop.core.variables.Variables;
+import org.apache.hop.junit.rules.RestoreHopEnvironment;
+import org.junit.*;
```
Comment (hansva) [inline plugins/databases/cratedb/src/test/java/org/apache/hop/databases/cratedb/CrateDBValueMetaBaseTest.java:38]: StarImports will fail with our new checkstyle configuration

File: plugins/databases/cratedb/src/test/java/org/apache/hop/databases/cratedb/CrateDBValueMetaBaseTest.java [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+import org.mockito.Spy;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
```
Comment (hansva) [inline plugins/databases/cratedb/src/test/java/org/apache/hop/databases/cratedb/CrateDBValueMetaBaseTest.java:48]: idem

File: plugins/databases/cratedb/src/test/java/org/apache/hop/databases/cratedb/CrateDBValueMetaBaseTest.java [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.*;
```
Comment (hansva) [inline plugins/databases/cratedb/src/test/java/org/apache/hop/databases/cratedb/CrateDBValueMetaBaseTest.java:51]: idem

**Review by hansva — CHANGES_REQUESTED** [review]

**Review by hansva — APPROVED** [review]

**Comment by bamaer** [comment]
@xdelox would you mind adding the cratedb.adoc page to [nav.adoc](https://github.com/apache/hop/blob/main/docs/hop-user-manual/modules/ROOT/nav.adoc) too please?

**Comment by xdelox** [comment]
> @xdelox would you mind adding the cratedb.adoc page to [nav.adoc](https://github.com/apache/hop/blob/main/docs/hop-user-manual/modules/ROOT/nav.adoc) too please?

oh, I missed this out! now it's done ✅

**Pull request merged**